src/lib/keys.js

```diff
@@ -34,7 +34,7 @@ Object.keys( Keys ).forEach( key => Keys[ key.toUpperCase() ] = Keys[ key ] );
 });
 
 // fn keys
-Array( 12 ).fill().forEach( ( item, index ) => Keys[ `f${ index + 1 }` ] = 112 + index );
+Array.apply(null, {length: 12}).map(Number.call, Number).forEach( ( item, index ) => Keys[ `f${ index + 1 }` ] = 112 + index );
```

Comment:
Sure, just an array is more simple, but given that you'd gone for a
programmatic way with `.fill` I thought you'd prefer this one :D It's not
my cleverness though, that's StackOverflow helping me out!

I'll update the PR with a simple array shortly.

On Thu, 21 Jul 2016, 23:30 Jed Verity <notifications@github.com> wrote:

> In src/lib/keys.js
> <https://github.com/glortho/react-keydown/pull/26#discussion_r71704532>:
>
> > @@ -34,7 +34,7 @@ Object.keys( Keys ).forEach( key => Keys[ key.toUpperCase() ] = Keys[ key ] );
> >  });
> >
> >  // fn keys
> > -Array( 12 ).fill().forEach( ( item, index ) => Keys[ `f${ index + 1 }` ] = 112 + index );
> > +Array.apply(null, {length: 12}).map(Number.call, Number).forEach( ( item, index ) => Keys[ `f${ index + 1 }` ] = 112 + index );
>
> @elyobo <https://github.com/elyobo> Thank you for this contribution and
> clever solution! However, I found it a bit hard to understand what was
> happening here at first. Given that we're producing side effects anyway we
> might as well switch to a for loop from 1-12 or something like that. What
> do you think?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/glortho/react-keydown/pull/26/files/dadc23ab5933c3ae1371ca48e8c7090d2ddcb315#r71704532>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAY4_jlIcDhPptcg5wcw1BqvwqQwO9FPks5qX3RjgaJpZM4JRgwR>
> .
>